modal/functions.py

```diff
@@ -1241,6 +1241,8 @@ def local(self, *args, **kwargs) -> Any:
         # TODO(erikbern): it would be nice to remove the nowrap thing, but right now that would cause
         # "user code" to run on the synchronicity thread, which seems bad
         info = self._get_info()
+        if is_local() and self.spec.volumes or self.spec.network_file_systems:
+            warnings.warn("Cannot mount volumes or file systems locally")
```

Comment:
I'd suggest a rewording of this error message:

- Can we use the `info` to include the name of the function that is being called? It will help users diagnose the issue
- I would explain a little bit more about what is happening — i.e. this sounds like the message you'd get for a hard error, but what's happening here is slightly different.

Maybe something like:

> The {function} function is executing locally and will not have access to the mounted Volume or NetworkFileSystem data